### PR TITLE
feat: switch to async await

### PR DESCRIPTION
**Pull request by achingbrain**
Converts the importer to `async`/`await`

- [x] Shim importer with `pull-stream-to-async-iterator`
- [x] Convert tests to `async`/`await`
- [x] Convert importer to `async`/`await` and remove shim
- [x] Update README with API changes

Requires ipfs/js-ipfs-unixfs-exporter#21 to be merged

### CODE REVIEW

**Comment by achingbrain** [comment]
Ref: ipfs/js-ipfs#1670

File: src/utils/persist.js [hunk 1]
```diff
@@ -22,10 +25,14 @@ const persist = (node, ipld, options, callback) => {
     codec = 'raw'
   }
 
-  if (hashAlg !== 'sha2-256') {
+  if (hashAlg !== 'sha2-256' && hashAlg !== mh.names['sha2-256']) {
```
Comment (alanshaw) [inline src/utils/persist.js]: How come this needs both checks?

File: test/importer-flush.spec.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ const map = require('pull-stream/throughs/map')
 const collect = require('pull-stream/sinks/collect')
 const pushable = require('pull-pushable')
 
-describe('importer: flush', () => {
+describe.skip('importer: flush', () => {
```
Comment (alanshaw) [inline test/importer-flush.spec.js]: skip?

**Review by alanshaw — APPROVED** [review]
Looks good so far!

File: src/utils/persist.js (hunk 1 above)
Comment (achingbrain) [reply src/utils/persist.js]: Oops, fixed.

File: test/importer-flush.spec.js (hunk 2 above)
Comment (achingbrain) [reply test/importer-flush.spec.js]: There's no flushing any more as it only seemed to be called by the tests.  Have removed them now.

**Pull request merged**